webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/design/SiteDesignActionHandler.java

```diff
@@ -0,0 +1,660 @@
+
+          /* verif que le nom de la page principale est correcte */
+    boolean searchOk = false;
+    Collection<File> collPages = controller.getAllWebPages2(currentPath);
+    for (File f : collPages) {
+      if (f.getName().equals(nomPage)) {
+        searchOk = true;
+        break;
+      }
+    }
+
+    SiteDetail descriptionSite2 =
+        new SiteDetail(id, controller.getComponentId(), nomSite, description, nomPage, type, null, null,
+            Integer.parseInt(etat), popup);
+
+    if (searchOk) {
+            /* update description en BD */
+      controller.updateWebSite(descriptionSite2);
+
+      if (listIcons.size() > 0) {
+        controller.associateIcons(id, listIcons);
+      }
+    } else {
+      request.setAttribute("SearchOK", Boolean.FALSE);
+      request.setAttribute(WebSitesRequestRouter.ICON_LIST_PARAM, listeIcones);
+    }
+
+    descriptionSite2 = controller.getWebSite(id);
+    controller.setSessionSite(descriptionSite2);
+    request.setAttribute("Site", descriptionSite2);
+  }
+
+  private String addNewDesignedSite(final WebSiteSessionController controller, final HttpRequest request)
+      throws WebSitesException {
+    // Filled at first access then null
+    String nomSite = request.getParameter(WebSitesRequestRouter.SITE_NAME_PARAM);
+    // Filled at first access then null
+    String description = request.getParameter(WebSitesRequestRouter.DESCRIPTION_PARAM);
+    // Filled at first access then null
+    String nomPage = request.getParameter(WebSitesRequestRouter.PAGE_NAME_PARAM);
+    String tempPopup = request.getParameter(WebSitesRequestRouter.POPUP_PARAM);
+    // Filled at first creation then null
+    String listeIcones = request.getParameter(WebSitesRequestRouter.ICON_LIST_PARAM);
+    // = en cas de new Site ou de classifySite
+    String listeTopics = request.getParameter(WebSitesRequestRouter.TOPIC_LIST_PARAM);
+    // Retrieve positions
+    String positions = request.getParameter("Positions");
+
+    int popup = 0;
+    if ((tempPopup != null) && (tempPopup.length() > 0)) {
+      popup = 1;
+    }
+
+    ArrayList<String> listIcons = new ArrayList<>();
+    if (listIcons != null) {
+      for (String icon : listeIcones.split(",")) {
+        if (!icon.isEmpty()) {
+          listIcons.add(icon);
+        }
+      }
+    }
+
+          /* recuperation de l'id */
+    String id = controller.getNextId();
+
+          /* Creer le repertoire id */
+    controller.createFolder(controller.getWebSitePathById(id));
+
+    // Persist siteDetail inside database type 0 = site cree
+    SiteDetail descriptionSite =
+        new SiteDetail(id, controller.getComponentId(), nomSite, description, nomPage, 0, null, null, 0,
+            popup);
+    descriptionSite.setPositions(positions);
+
+    String pubId = controller.createWebSite(descriptionSite);
+    descriptionSite = controller.getWebSite(id);
+    controller.setSessionSite(descriptionSite);
+
+    if (listIcons.size() > 0) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1155)